### PR TITLE
Backports from master

### PR DESCRIPTION
**Pull request by vicentebolea**
Backported Pull Requests

#4181  EVPath Upstream                            eisenhauer:EVup                       about 19 hours ago
#4169  Add API call to reset memory selection     franzpoeschel:reset-memory-selection  about 7 days ago
#4166  Typo in documentation                      anagainaru:typo-build-doc             about 11 days ago
#4161  Adding hooks to gather performance for...  anagainaru:derived-performance        about 17 days ago
#4153  Minor docs tweaks                          eisenhauer:DocsUpdate                 about 1 month ago
#4147  ReadModeFlatten                            eisenhauer:Flatten                    about 1 month ago
#4142  pugixml,cmake: fix linkage when pugixm...  vicentebolea:fix-pugixml-dep          about 1 month ago
#4138  Fixes for FreeBSD, including upstream      eisenhauer:FreeBSD                    about 1 month ago
#4185  Kill outdated reference to bindings.Mode.ReadFlattenSteps

### CODE REVIEW

**Comment by eisenhauer** [comment]
I think this is missing #4160, which removes some stuff that was added in #4147 in favor of another approach...

File: python/adios2/stream.py
Comment (eisenhauer) [inline python/adios2/stream.py:1]: Unfortunate.  All references to Mode ReadFlattenSteps should have been removed.  Looks like they still exist in python.  This probably requires a new PR.

Comment (vicentebolea) [reply python/adios2/stream.py:1]: is this blocking our patch release?

Comment (eisenhauer) [reply python/adios2/stream.py:1]: Actually, the reference here, bindings.Mode.ReadFlattenSteps has been removed.  So, this is clearly erroneous, but CI works fine.  When would this have failed?

Comment (vicentebolea) [reply python/adios2/stream.py:1]: In python if that line is not executed then it does not complains. (duck typing)

Comment (eisenhauer) [reply python/adios2/stream.py:1]: Ah.  Well, it's an easy fix.  I've submitted the PR.

**Comment by vicentebolea** [comment]
@eisenhauer it appears that we are not ABI compatible anymore at least since #4144. Does it make sense to revert it? releasing a non abi compatible patch release might result in trouble for conda users (or other distros).

**Comment by vicentebolea** [comment]
No worries, I was able to come up with a ABI compatible fix.

**Comment by eisenhauer** [comment]
Excellent...

**Review by eisenhauer — APPROVED** [review]
I don't see further problems...

**Review by eisenhauer — APPROVED** [review]

**Comment by vicentebolea** [comment]
Screenshot from 2024-05-28 19-49-21


Screenshot from 2024-05-28 19-49-12


Passing :smile:

**Comment by eisenhauer** [comment]
> Passing 😄

Great!

**Pull request merged**